### ISSUE TITLE
Issues in Date Validation

### ISSUE THREAD

**Issue by sanketrelfor** [tdegrunt/jsonschema#244]
```javascript
'use strict';

const Validator = require('jsonschema').Validator;

const schema1 = {
  'id': 'test',
  'type': 'object',
  'additionalProperties': false,
  'properties': {
    'birthDate': {
      'type': 'string', 
      'format': 'date'
    }
  },
  'required': [
    'birthDate'
  ]
};

const all = {

  birthDate: '2018-02-31'
};
const v = new Validator();
const res = v.validate(all, schema1);
console.log(res.errors);

```

Above code is giving provided date as valid one, please look into this.

**Comment by nbering** [comment]
You are correct, this is invalid according to the [JSON schema specification](http://json-schema.org/latest/json-schema-validation.html#rfc.section.7.3.1), which takes it's date format from [RFC3339 section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6).

The fault lies with the regular expression, here:
https://github.com/tdegrunt/jsonschema/blob/038fc524548ae19fd6fa5fb88bb08b9fea087ab6/lib/helpers.js#L117

The RFC3339 specification does indicate that days shall be validated against the month and year. However, and simple regular expression is unsuitable for validating such a complex requirement, especially when accounting for the complex logic associated with leap years.

**Comment by awwright** [comment]
@sanfords This should be fixed in master branch. Which version are you using, can you tell?

**Comment by sanfords** [comment]
@awwright, Me?

**Comment by awwright** [comment]
Wait what, I'm sorry

**Comment by awwright** [comment]
I'm going to fix this as a part of #257
